src/components/ScenarioParameters/components/ScenarioParametersTabs/ScenarioParameterInput.js

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React from 'react';
+import { DATASET_ID_VARTYPE } from '../../../../services/config/ApiConstants';
+import { ConfigUtils } from '../../../../utils/ConfigUtils';
+import { VAR_TYPES_MAPPING } from '../../../../utils/scenarioParameters/VarTypesMapping';
+import { useTranslation } from 'react-i18next';
+import PropTypes from 'prop-types';
+
+const ScenarioParameterInput = ({ datasets, parameterData, parametersState, setParametersState, editMode }) => {
+  const { t } = useTranslation();
+  const parameterVarType = ConfigUtils.buildExtendedVarType(parameterData.varType, parameterData.subType);
+  let varTypeFactory;
+
+  if (parameterVarType in VAR_TYPES_MAPPING) {
+    varTypeFactory = VAR_TYPES_MAPPING[parameterVarType];
+  } else {
+    varTypeFactory = VAR_TYPES_MAPPING[parameterData.varType];
+  }
+
+  if (varTypeFactory === undefined) {
+    console.warn('No factory defined for varType ' + parameterVarType);
+    return null;
+  }
+  if (varTypeFactory === null) {
+    return null;
+  }
+
+  const props = { parameterData, parametersState, setParametersState, editMode, key: parameterData.id };
+  if (parameterVarType.startsWith(DATASET_ID_VARTYPE)) {
+    props.datasets = datasets;
+  }
+  // name property helps distinguish React components from factories
+  if ('name' in varTypeFactory) {
+    return React.createElement(varTypeFactory, props);
+  }
+  return varTypeFactory.create(t, parameterData, parametersState, setParametersState, editMode);
```

Comment:
Could we add a warning after the `if` block to warn users that the "function-shape" factories are now deprecated and won't be supported anymore in a future version ?